### ISSUE TITLE
Sklearn 1.9 is not compatible with ZeroShotClassification

### ISSUE THREAD

**Issue by Samoed** [embeddings-benchmark/mteb#4784]
### Describe the bug

Changes that break zeroshot classification that passed labels as strings mostly
> Fix [metrics.accuracy_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.accuracy_score.html#sklearn.metrics.accuracy_score), [metrics.hamming_loss](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.hamming_loss.html#sklearn.metrics.hamming_loss) [metrics.zero_one_loss](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.zero_one_loss.html#sklearn.metrics.zero_one_loss), [metrics.matthews_corrcoef](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.matthews_corrcoef.html#sklearn.metrics.matthews_corrcoef) and [metrics.confusion_matrix](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html#sklearn.metrics.confusion_matrix) (when labels is not None) now raise an error when y_true is string and y_pred is numeric, for all array-like inputs. Previously, lists and numpy arrays not of object dtype did not raise an error for this mixed input case. The above metrics will also raise an error for [label indicator matrix](https://scikit-learn.org/stable/glossary.html#term-label-indicator-matrix) inputs of inconsistent size, except for [metrics.confusion_matrix](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html#sklearn.metrics.confusion_matrix) which does not accept label indicator matrix inputs. By [Lucy Liu](https://github.com/lucyleeow). [#33086](https://github.com/scikit-learn/scikit-learn/pull/33086) https://scikit-learn.org/stable/whats_new/v1.9.html#sklearn-metrics

### To reproduce

Run any zeroshot 

### Are you interested to contribute a fix for this bug?

Yes

**Comment by KennethEnevoldsen** [comment]
remove bug as it is no longer a bug, but def. a high priority fix
